includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -755,6 +761,41 @@ private function has_used_attributes( $attribute_names ) {
 		return true;
 	}
 
+	/**
+	 * Whether a given class is allowed to be styled in <amp-date-picker>.
+	 *
+	 * That component has child classes that won't be present in the document yet.
+	 * So get whether a class is an allowed child.
+	 *
+	 * @since 1.5.0
+	 * @link https://github.com/airbnb/react-dates/tree/05356/src/components
+	 *
+	 * @param string $class The name of the class to evaluate.
+	 * @return bool Whether the class is allowed as a child of <amp-date-picker>.
+	 */
+	private function is_class_allowed_in_amp_date_picker( $class ) {
+		// Some prefixes, some full class names.
+		$allowed_class_beginnings = [
+			'CalendarDay',
+			'CalendarMonth',
+			'DateInput',
+			'DateRangePicker',
+			'DayPicker',
+			'KeyboardShortcutRow',
+			'SingleDatePicker',
+			'amp-date-picker-calendar-container',
```

Comment:
If needed, this check for class names could be more specific. For example, this class isn't a prefix, it should be an exact match of the class.

Most other classes in this array are prefixes. For example, `DayPicker` is a prefix of `DayPicker_focusRegion`.

Reply:
I'm going to make a commit for this.

Reply:
Applied in https://github.com/ampproject/amp-wp/pull/4339/commits/28d037e1d92c5f1404e53adbaa951f237622ac75